### ISSUE TITLE
Error with User Names -related to phone numbers

### ISSUE THREAD

**Issue by jopowers** [megandalster/rmhp-homebase#38]
Hi Allen,

We have slowly begun to introduce Homebase to our volunteers, we decided to do a very slow introduction and roll-out to get them all acclimated.   We have had several volunteers that needed to change their phone numbers. An example is Ann Turrell -she changed her primary number after her first log in. She was able to get in and out of Homebase but I noticed there is now an error showing up under her name in the Master Schedule. We have had this happen with several other volunteers as well and we have had to delete the record and reset the password.  Is there a way to simplify this issue as we anticipate numbers changing often, especially since Sue and I did all the data input.

**Comment by megandalster** [comment]
Hi Joanna,

This is a weakness of Homebase, which is built on the idea that a person’s first name + primary phone number provides a unique key for locating that person in the database.  As you know, the master schedule and calendar rely on that uniqueness.

For the Portland House, I changed the Homebase volunteer form so that a person cannot change her first name or primary phone number.  If her phone number actually changes, she must contact the volunteer coordinator, who removes and re-addeds the volunteer to the database (I just did that for Ann Turrell).  Alternatively, the volunteer can herself record her new phone number as her “secondary phone number”, leaving her primary phone number unchanged.  That choice creates other problems, so there is no perfect solution.

I can do for you what I did for the Portland House, forcing the volunteer to contact you if she really needs to change her primary phone number.  When that happens, your making the deletion and addition will show up in your home page log, which will allow you to replace that person in the master schedule and calendar as well.  I know of no easier fix for this problem.

Just curious, are the “several volunteers" needing to change their phone numbers mostly because the numbers are wrong in Homebase or mostly because they really changed their phone numbers?  My sense (hope) is that people don’t often change their primary phone numbers so that once your volunteer database settles down this issue won’t happen too often.

Allen

On Oct 20, 2015, at 11:12 AM, jopowers <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Allen,

We have slowly begun to introduce Homebase to our volunteers, we decided to do a very slow introduction and roll-out to get them all acclimated. We have had several volunteers that needed to change their phone numbers. An example is Ann Turrell -she changed her primary number after her first log in. She was able to get in and out of Homebase but I noticed there is now an error showing up under her name in the Master Schedule. We have had this happen with several other volunteers as well and we have had to delete the record and reset the password. Is there a way to simplify this issue as we anticipate numbers changing often, especially since Sue and I did all the data input.

**Comment by jopowers** [comment]
Hi Allen,

Thanks for the explanation, we have deleted a few records, as you suggested. Because Sue and I entered the data off of the volunteer applications we are anticipating a few "hiccups".  For example in Ann's case I filled in the database according to her application, however she does not use her primary home number anymore and wanted it to be her cell as the primary. We also have some volunteers that have applications that are over 10 years old and I can anticipate some folks switching over to their cell phone as their primary, since so many people don't use their landlines anymore (myself included). We will proceed with deleting if necessary.

However, we have noticed that our volunteers don't seem to have access to their application to make any changes, ie address, phone number, etc. Is this correct? I was a bit confused when you said the Portland house volunteers are able make changes.

**Comment by megandalster** [comment]
Right, the Portland volunteers have access to their own (and only their own) profile data, so that they can update it as needed.  The link to their data page would be from their home page after logging in.

I can do this for you if you wish — that would distribute your record-keeping work a bit.  If we do this, I would recommend that volunteers not be able to change their own first names or phone numbers.  That would remain in your hands.  Let me know.  Allen

On Oct 20, 2015, at 4:50 PM, jopowers <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Allen,

Thanks for the explanation, we have deleted a few records, as you suggested. Because Sue and I entered the data off of the volunteer applications we are anticipating a few "hiccups". For example in Ann's case I filled in the database according to her application, however she does not use her primary home number anymore and wanted it to be her cell as the primary. We also have some volunteers that have applications that are over 10 years old and I can anticipate some folks switching over to their cell phone as their primary, since so many people don't use their landlines anymore (myself included). We will proceed with deleting if necessary.

However, we have noticed that our volunteers don't seem to have access to their application to make any changes, ie address, phone number, etc. Is this correct? I was a bit confused when you said the Portland house volunteers are able make changes.
